src/components/core/utils/validateDdoHandler.ts

```
@@ -0,0 +1,155 @@
+    .filter((result: any) => result.path)
+    .map((result: any) => toNT(result.path))
+    .map((path: any) => path.replace('http://schema.org/', ''))
+
+  const messages = results
+    .filter((result: any) => result.message)
+    .map((result: any) => toNT(result.message))
+    .map(beautifyMessage)
+
+  return Object.fromEntries(
+    paths.map((path: string, index: number) => [path, messages[index]])
+  )
+}
+
+function beautifyMessage(message: string): string {
+  if (message.startsWith('Less than 1 values on')) {
+    const index = message.indexOf('->') + 2
+    message = 'Less than 1 value on ' + message.slice(index)
+  }
+  return message
+}
+
+function isIsoFormat(dateString: string): boolean {
+  const isoDateRegex = /^\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}(\.\d{1,3})?Z)?$/
+  return isoDateRegex.test(dateString)
+}
+
+function makeDid(nftAddress: string, chainId: string): string {
+  return (
+    'did:op:' +
+    createHash('sha256')
+      .update(getAddress(nftAddress) + chainId)
+      .digest('hex')
+  )
+}
+export async function validateObject(
+  obj: Record<string, any>,
+  chainId: number,
+  nftAddress: string
+): Promise<[boolean, Record<string, string>]> {
+  const ddoCopy = obj
+  ddoCopy['@type'] = 'DDO'
+  const extraErrors: Record<string, string> = {}
+  if (!('@context' in obj)) {
+    extraErrors['@context'] = 'Context is missing.'
+  }
+  if ('@context' in obj && !Array.isArray(obj['@context'])) {
+    extraErrors['@context'] = 'Context is not an array.'
+  }
+  if (!('metadata' in obj)) {
+    extraErrors.metadata = 'Metadata is missing or invalid.'
+  }
+
+  ;['created', 'updated'].forEach((attr) => {
+    if ('metadata' in obj && attr in obj.metadata && !isIsoFormat(obj.metadata[attr])) {
+      extraErrors.metadata = `${attr} is not in ISO format.`
+    }
+  })
+
+  if (!chainId) {
+    extraErrors.chainId = 'chainId is missing or invalid.'
+  }
+
+  if (!nftAddress || nftAddress === '' || !isAddress(nftAddress.toLowerCase())) {
```

there are two useful functions isAddress and getAddress
it is better to use getAddress because it does both checks and transformation to the standard form of address in general
keep in mind that if the address is bad, it will throw an error rather than return false or null
https://docs.ethers.org/v6/api/address/#getAddress

---

Sure, thank you!